### ISSUE TITLE
[🔨 compilation issue] memset compile warnings

### ISSUE THREAD

**Issue by nyue** [mitsuba-renderer/mitsuba3#141]
``` [236/243] Building CXX object src/textures/CMakeFiles/bitmap.dir/bitmap.cpp.o
In member function ‘void drjit::Texture<Value_, Dimension_>::init(const size_t*, size_t, bool, drjit::FilterMode, drjit::WrapMode, bool) [with Value = float; long unsigned int Dimension = 2]’:
cc1plus: warning: ‘void* __builtin_memset(void*, int, long unsigned int)’ specified bound 18446744073709551612 exceeds maximum object size 9223372036854775807 [-Wstringop-overflow=]
[242/243] Building CXX object src/volumes/CMakeFiles/gridvolume.dir/grid.cpp.o
In member function ‘void drjit::Texture<Value_, Dimension_>::init(const size_t*, size_t, bool, drjit::FilterMode, drjit::WrapMode, bool) [with Value = float; long unsigned int Dimension = 3]’:
cc1plus: warning: ‘void* __builtin_memset(void*, int, long unsigned int)’ specified bound 18446744073709551612 exceeds maximum object size 9223372036854775807 [-Wstringop-overflow=]
[243/243] Linking CXX shared library plugins/gridvolume.so
```

## Summary

String overflow compilation warning may become potential security issue, hence reporting it.

## System configuration

*For bug report, please enter information regarding your system configuration<- [remove this]*

- Platform: ...
- Ubuntu 18.04
- Compiler: ...
- GCC 9.3.0
- Python version: ...
- Not building with Python integration, just C++
- Mitsuba 3 version: ...
- Compiled variants:
    * `scalar_rgb`
    * ...

## Description

*Please enter the full description of your issue <- [remove this]*

## Steps to reproduce

*For bug report, please enumerate the steps to follow in order to reproduce the issue <- [remove this]*

1. ...
2. ...

**Comment by njroussel** [comment]
Hi @nyue 
Thanks for reporting this.

It seems to actually be a warning caused in DrJit, which bizarrely doesn't show the same error when compiled by itself. I'll keep this issue open until we fix it over there.

**Comment by Speierers** [comment]
Hi @nyue ,

Thanks to @njroussel 's investigation, this should now be fixed in the next release: [8f2ebfe](https://github.com/mitsuba-renderer/drjit/commit/8f2ebfedf44fcc193291605173b2eded87fd6966)
